lib/kubeclient/common.rb

```diff
@@ -202,20 +202,16 @@ def delete_entity(entity_type, name, namespace = nil)
     end
 
     def create_entity(entity_type, entity_config, klass)
-      # to_hash should be called because of issue #9 in recursive open
-      # struct
-      hash = entity_config.to_hash
-
-      ns_prefix = build_namespace_prefix(entity_config.metadata['table'][:namespace])
+      ns_prefix = build_namespace_prefix(entity_config.metadata.namespace)
 
       # TODO: temporary solution to add "kind" and apiVersion to request
       # until this issue is solved
       # https://github.com/GoogleCloudPlatform/kubernetes/issues/6439
-      hash['kind'] = entity_type
-      hash['apiVersion'] = @api_version
+      entity_config['kind'] = entity_type
+      entity_config['apiVersion'] = @api_version
```

Comment:
@bkoski you haven't mentioned this change in behavior: you're now changing the original object passed in (`entity_config`).